### PR TITLE
Sample/Test syntax changes and sanitycheck script cleanup

### PR DESCRIPTION
**Pull request by nashif**
-  boards: fix board yaml syntax
-  tests/samples: fixed yaml syntax
-  sanitycheck: refactor data parsing in script
-  sanitycheck: flake8 fixes and move closer to pep8
-  sanitycheck: use test instead of section
-  sanitycheck: reworked Platform/Architecture classes
-  sanitycheck: remove platform-level option
-  sanitycheck: we do not need to pass ARCH= on command line
-  sanitycheck: descriptive variable names
-  sanitycheck: remove usage of qemu for generic handlers
-  sanitycheck: disable expr parser debug
-  sanitycheck: allow placing parsetab.py in a custom dir

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

File: samples/basic/threads/sample.yaml [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
+    using K_THREAD_DEFINE
   name: Basic Thread Demo
 tests:
-  - test:
-      build_only: true
-      platform_whitelist: 96b_carbon
-      tags: samples
+  test:
+    build_only: true
+    platform_whitelist: 96b_carbon
```
Comment (andrewboie) [inline samples/basic/threads/sample.yaml]: why is this filtered like this?

File: samples/drivers/crypto/sample.yaml [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-      extra_args: CONF_FILE=prj_mtls_shim.conf
-      tags: crypto
+  test-mbedtls:
+    build_only: true
+    extra_args: CONF_FILE=prj_mtls_shim.conf
+    platform_whitelist: qemu_x86
+    tags: crypto
+  test-micro:
+    build_only: true
+    platform_whitelist: qemu_x86
```
Comment (andrewboie) [inline samples/drivers/crypto/sample.yaml]: why is this filtered?

File: scripts/sanitycheck [hunk 3]
```diff
@@ -1310,19 +1307,15 @@ class TestSuite:
...
                 workdir = os.path.relpath(dirpath, testcase_root)
 
-                common = {}
-                if 'common' in cp.cp:
-                    common = cp.cp['common']
-
-                for section in cp.sections():
-                    name = list(section.keys())[0]
-                    tc_dict = cp.get_section(name, testcase_valid_keys, common)
+                print(parsed_data.tests)
```
Comment (andrewboie) [inline scripts/sanitycheck]: get rid of this or make it a debug() imo

**Review by tyrantlizard77 — COMMENTED** [review]
At some point I think we really need to audit our use of platform_whitelist, but doesn't need to be in this PR.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by nashif** [comment]
@andrewboie I agree on the whitelist comment, I have ton of changes on top of this cleaning up the build_only tag and the whitelisting, this one was just for cleaning up the syntax and the sanitycheck script as a whole.

**Pull request merged**